### PR TITLE
[DataGrid] Fix role=presentation a11y issue

### LINKED ISSUE #12837: [data grid] AXE test fails due to invalid child role "presentation"
### Steps to reproduce

Link to live example: https://mui.com/x/react-data-grid/

Steps:
1. Open the MUI demo page for data grid or data grid pro
2. Run Axe test with Axe Chrome plugin
3. Note errors:
```
Element has children which are not allowed: [role=presentation]
Element uses aria-busy="true" while showing a loader
```
Screenshot 2024-04-18 at 11 18 42 AM

### Current behavior

DataGrid fails accessibility test out of the box

### Expected behavior

Should not fail accessibility test

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
 Using Chrome Version 124.0.6367.62
System:
    OS: macOS 14.4.1
  Binaries:
    Node: 20.12.2 - ~/.nvm/versions/node/v20.12.2/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v20.12.2/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: 124.0.6367.62
    Edge: Not Found
    Safari: 17.4.1
  npmPackages:
    @emotion/react: 11.11.1 => 11.11.1 
    @emotion/styled: 11.11.0 => 11.11.0 
    @mui/base: ^5.0.0-beta.24 => 5.0.0-beta.40 
    @mui/core-downloads-tracker:  5.15.15 
    @mui/icons-material: ^5.15.15 => 5.15.15 
    @mui/lab: ^5.0.0-alpha.124 => 5.0.0-alpha.170 
    @mui/material: ^5.15.15 => 5.15.15 
    @mui/private-theming:  5.15.14 
    @mui/styled-engine:  5.15.14 
    @mui/styled-engine-sc: ^5.14.12 => 5.14.12 
    @mui/system: ^5.15.14 => 5.15.15 
    @mui/types: ^7.2.14 => 7.2.14 
    @mui/utils:  5.15.14 
    @mui/x-data-grid:  7.2.0 
    @mui/x-data-grid-generator: ~7.2.0 => 7.2.0 
    @mui/x-data-grid-premium:  7.2.0 
    @mui/x-data-grid-pro: ~7.2.0 => 7.2.0 
    @mui/x-date-pickers: ~7.2.0 => 7.2.0 
    @mui/x-date-pickers-pro: ~7.2.0 => 7.2.0 
    @mui/x-license: ^7.2.0 => 7.2.0 
    @mui/x-tree-view: ~7.2.0 => 7.2.0 
    @types/react: ^18.2.79 => 18.2.79 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    styled-components: ^5.3.11 => 5.3.11 
    typescript: 5.3.3 => 5.3.3 

```
</details>

**Search keywords**: invalid child role aria
**Order ID**: Order #75258

### PR DESCRIPTION
**Pull request by romgrk**
Closes #12837 

The `tabIndex=-1` on this element is probably a leftover from a refactor as the element doesn't need to be keyboard-focusable.

### CODE REVIEW

File: packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx [hunk 1]
```diff
@@ -583,7 +583,6 @@ export const useGridVirtualScroller = () => {
     }),
     getScrollerProps: () => ({
       ref: scrollerRef,
-      tabIndex: -1,
```
Comment (cherniavskii) [inline packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: Without the explicit tabIndex here, Firefox focuses on the virtual scroller element:

### Chromium:
https://github.com/user-attachments/assets/3b0444da-ecf7-4619-bdf5-166744c471cb

### Firefox:
https://github.com/user-attachments/assets/406ab2a9-78c8-4832-aa52-8c168711128c

Comment (samwato) [reply packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: Does the Scrollbar may also need `tabIndex: -1` to be removed?

Screenshot from 2024-07-19 10-07-35

Comment (romgrk) [reply packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: I'll look into the scrollbar.

@cherniavskii Wouldn't that be a firefox issue? That element is not listening to any key event, only to wheel/scroll ones. I could restore the tabIndex only for FF and we file a bug on their tracker?

Comment (romgrk) [reply packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: I've added `aria-hidden=true` for the scrollbar, same as ag-grid does, it's probably the best aria tag for that case.

@cherniavskii For the Firefox issue, wouldn't that be a Firefox bug? Can you test on Safari to see what's the behavior? If Chrome & Safari agree and Firefox disagrees, we could add the tabindex only as a workaround for FF and file an issue with them.
Looks vaguely similar to this: https://bugzilla.mozilla.org/show_bug.cgi?id=1069739

Comment (cherniavskii) [reply packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: I checked Chrome, Safari, and Firefox, and the issue is only reproducible in Firefox.

> we could add the tabindex only as a workaround for FF and file an issue with them

Yeah, that makes sense 👍🏻

> Looks vaguely similar to this: [bugzilla.mozilla.org/show_bug.cgi?id=1069739](https://bugzilla.mozilla.org/show_bug.cgi?id=1069739)

 😅

Comment (romgrk) [reply packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:586]: Pretty common lifespan for a firefox bug, they've been around for longer and have way less resources :|

**Pull request merged**